pytorch_tabnet/utils.py

```diff
@@ -190,3 +202,57 @@ def filter_weights(weights):
     if isinstance(weights, dict):
         raise ValueError(err_msg + "Dict given.")
     return
+
+
+def validate_eval_set(eval_set, eval_name, X_train, y_train):
+    """Check if the shapes of eval_set are compatible with (X_train, y_train).
+
+    Parameters
+    ----------
+    eval_set: list of tuple
+        List of eval tuple set (X, y).
+        The last one is used for early stopping
+    eval_names: list of str
+        List of eval set names.
+    X_train: np.ndarray
+        Train owned products
+    y_train : np.array
+        Train targeted products
+
+    Returns
+    -------
+    eval_names : list of str
+        Validated list of eval_names.
+    eval_set : list of tuple
+        Validated list of eval_set.
+
+    """
+    eval_set = eval_set or []
+    eval_name = eval_name or [f"val_{i}" for i in range(len(eval_set))]
+
+    assert len(eval_set) == len(
+        eval_name
+    ), "eval_set and eval_name have not the same length"
+    if len(eval_set) > 0:
+        assert all(
+            len(elem) == 2 for elem in eval_set
+        ), "Each tuple of eval_set need to have two elements"
```

Comment:
Does that mean that you can't compute multiple metrics with one DataLoader? This will slow down the evaluation quite a lot if you need to use different dataloaders everytime, the most common use case is to have different metrics on the same validation set no?

Reply:
It just to check if all element into the eval_set have a X and Y (2 elements)

Reply:
Maybe my question was not clear : is it possible to ask for multiple metrics on the same eval set?